cmake/modules/grpc.cmake

```diff
@@ -3,30 +3,55 @@ option(USE_BUNDLED_GRPC "Enable building of the bundled grpc" ${USE_BUNDLED_DEPS
 if(GRPC_INCLUDE)
 	# we already have grpc
 elseif(NOT USE_BUNDLED_GRPC)
-	find_library(GPR_LIB NAMES gpr)
-	if(GPR_LIB)
-		message(STATUS "Found gpr lib: ${GPR_LIB}")
-	else()
-		message(FATAL_ERROR "Couldn't find system gpr")
-	endif()
-	find_path(GRPCXX_INCLUDE NAMES grpc++/grpc++.h)
-	if(GRPCXX_INCLUDE)
-		set(GRPC_INCLUDE ${GRPCXX_INCLUDE})
-	else()
-		find_path(GRPCPP_INCLUDE NAMES grpcpp/grpcpp.h)
-		set(GRPC_INCLUDE ${GRPCPP_INCLUDE})
-		add_definitions(-DGRPC_INCLUDE_IS_GRPCPP=1)
-	endif()
-	find_library(GRPC_LIB NAMES grpc)
-	find_library(GRPCPP_LIB NAMES grpc++)
-	if(GRPC_INCLUDE AND GRPC_LIB AND GRPCPP_LIB)
-		message(STATUS "Found grpc: include: ${GRPC_INCLUDE}, C lib: ${GRPC_LIB}, C++ lib: ${GRPCPP_LIB}")
+	# gRPC
+	find_package(gRPC CONFIG)
+	if(gRPC_FOUND)
+		message(STATUS "Using gRPC ${gRPC_VERSION}")
+		set(GPR_LIB gRPC::gpr)
+		set(GRPC_LIB gRPC::grpc)
+		set(GRPCPP_LIB gRPC::grpc++)
+
+		# # gRPC C++ plugin
+		get_target_property(GRPC_CPP_PLUGIN gRPC::grpc_cpp_plugin LOCATION)
+		if(NOT GRPC_CPP_PLUGIN)
+			message(FATAL_ERROR "System grpc_cpp_plugin not found")
+		endif()
+
+		# gRPC include dir + properly handle grpc{++,pp}
+		get_target_property(GRPC_INCLUDE gRPC::grpc++ INTERFACE_INCLUDE_DIRECTORIES)
+		find_path(GRPCXX_INCLUDE NAMES grpc++/grpc++.h)
+		if(NOT GRPCXX_INCLUDE)
+			find_path(GRPCPP_INCLUDE NAMES grpcpp/grpcpp.h)
+			add_definitions(-DGRPC_INCLUDE_IS_GRPCPP=1)
+		endif()
```

Comment:
Do we want to pass `PATHS ${GRPC_INCLUDE}` (or something that uses the pkg_config data) to the `find_path` calls here?

Reply:
I think we might; but that was there before my PR and i didn't want to touch it (gRPC cmake is a bit like the ultimate boss to fight against! Now i unlocked holiness badge :P)
I mean, that checking code is the same as before.
Btw we could adjust it, if you think it is needed :)

Reply:
yeah, the grpc build scripts are on a whole different level :D 

my reasoning for the requested change is:
* the old code always did find_library etc. without any hints so only searched the default paths, whatever they are
* the new code can potentially find grpc elsewhere, but we do:
    * get the GRPC_INCLUDE property from the grpc cmake module (good)
    * look for grpc++/grpc++.h in the *default locations* (bad)
    * if not found, look for grpcpp/grpcpp.h in the *default locations* again (bad too)

so, this new code appears more flexible but in the end still requires the grpc headers to live in the default location

... or I'm completely wrong :D

Reply:
Nope! The problem was that the old code **assumed** that grpc libraries had built-in all their deps, ie: statically linked.
The new code instead is able to correctly use a grpc that dynamically links its dependencies, while still managing statically linked one.  

The old case was left still because we found out that `find_package(gRPC)` can fail on some distro because gRPC cmake config module is not installed.

To summarize:
* package found -> both dynamically and statically linked gRPC work fine
* package not found -> current libs master behavior

> the new code can potentially find grpc elsewhere, but we do: 

BTW you are right, i will update that!

Reply:
Note however that
```
find_path(GRPCXX_INCLUDE NAMES grpc++/grpc++.h)
		if(NOT GRPCXX_INCLUDE)
			find_path(GRPCPP_INCLUDE NAMES grpcpp/grpcpp.h)
			add_definitions(-DGRPC_INCLUDE_IS_GRPCPP=1)
		endif()
```
is just used to set the GRPC_INCLUDE_IS_GRPCPP define ;)

Reply:
Yes, but that assumes that `grpc++/grpc++.h` can be found in the default include paths.

If `GRPC_INCLUDE` is non-standard, we'll still look for grpc++.h in the default locations, so we won't find it if it exists in e.g. /opt/grpc/include/grpc++.h` (where the grpc cmake pointed us to).

If we simply add `PATHS ${GRPC_INCLUDE}` to the `find_path` invocation, we should be good (I think).

Reply:
> the new code can potentially find grpc elsewhere, but we do:

Should be fixed now! Thanks for your input!